CDTDatastore/touchdb/TDPuller.m

```diff
@@ -398,8 +462,62 @@ - (void)pullRemoteRevision:(TD_Revision*)rev
     [dl start];
 }
 
+- (void)pullBulkRevisionsBulkGet:(NSArray*)bulkRevs
+{
+    // http://wiki.apache.org/couchdb/HTTP_Bulk_Document_API
+    NSUInteger nRevs = bulkRevs.count;
+    if (nRevs == 0) return;
+    CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@ bulk-fetching %u remote revisions...", self,
+               (unsigned)nRevs);
+    CDTLogVerbose(CDTREPLICATION_LOG_CONTEXT, @"%@ bulk-fetching remote revisions: %@", self, bulkRevs);
+    
+    [self asyncTaskStarted];
+    ++_httpConnectionCount;
+    
+    // body needs to be in form:
+    // {"docs":[{"id":"1-foo","rev":"rev123","atts_since":["1-foo,...]}]}
+    NSArray* keys = [bulkRevs my_map:^(TD_Revision* rev) {
+        NSArray* knownRevs = [_db getPossibleAncestorRevisionIDs:rev limit:kMaxNumberOfAttsSince];
+        return @{@"id": rev.docID,
+                 @"rev": rev.revID,
+                 @"atts_since": knownRevs != nil ? knownRevs : @[]};
+    }];
+    
+    NSDictionary *requestBody = @{@"docs": keys};    
+    NSMutableArray* remainingRevs = [bulkRevs mutableCopy];
+
+    [self sendAsyncRequest:@"POST"
+                      path:@"_bulk_get?revs=true&attachments=true"
+                      body:requestBody
+              onCompletion:^(id result, NSError* error) {
+
+                  // unpack revisions and queue them in _downloadsToInsert
+                  NSArray *results = result[@"results"];
```

Comment:
This does appear to be `nil` safe, but I agree that it could crash if `result` isn't an `NSDictionary`. Also, we don't seem to be handling any non-nil `error` passed into this completion block.